java/daikon/dcomp/DCInstrument.java

```diff
@@ -1858,17 +1949,19 @@ InstructionList handleInvoke(InvokeInstruction invoke) {
     if (invoke instanceof INVOKEDYNAMIC) {
```

Comment:
I suggest abstracting out the code from this line until the last assignment to `callee_instrumented`, into a method named something like `isCalleeInstrumented` or `isMethodInstrumented`.  This will make the main logic of the method much easier to follow, and will help to demarcate the different parts of this method.